# Titanic competition with TensorFlow Decision Forests

This notebook will take you through the steps needed to train a baseline Gradient Boosted Trees Model using TensorFlow Decision Forests and creating a submission on the Titanic competition. 

This notebook shows:

1. How to do some basic pre-processing. For example, the passenger names will be tokenized, and ticket names will be splitted in parts.
1. How to train a Gradient Boosted Trees (GBT) with default parameters
1. How to train a GBT with improved default parameters
1. How to tune the parameters of a GBTs
1. How to train and ensemble many GBTs

# Imports dependencies

In [18]:
import numpy as np
import pandas as pd
import os

import tensorflow as tf
import tensorflow_decision_forests as tfdf

print(f"Found TF-DF {tfdf.__version__}")

Found TF-DF 1.2.0


# Load dataset

In [19]:
train_df = pd.read_csv("/kaggle/input/titanic/train.csv")
serving_df = pd.read_csv("/kaggle/input/titanic/test.csv")

train_df.head(10)

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S
5,6,0,3,"Moran, Mr. James",male,NaN,0,0,330877,8.4583,NaN,Q
6,7,0,1,"McCarthy, Mr. Timothy J",male,54.0,0,0,17463,51.8625,E46,S
7,8,0,3,"Palsson, Master. Gosta Leonard",male,2.0,3,1,349909,21.0750,NaN,S
8,9,1,3,"Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)",female,27.0,0,2,347742,11.1333,NaN,S
9,10,1,2,"Nasser, Mrs. Nicholas (Adele Achem)",female,14.0,1,0,237736,30.0708,NaN,C


# Prepare dataset

We will apply the following transformations on the dataset.

1. Tokenize the names. For example, "Braund, Mr. Owen Harris" will become ["Braund", "Mr.", "Owen", "Harris"].
2. Extract any prefix in the ticket. For example ticket "STON/O2. 3101282" will become "STON/O2." and 3101282.

In [92]:
def preprocess(df):
    df = df.copy()
    
    def normalize_name(x):
        return " ".join([v.strip(",()[].\"'") for v in x.split(" ")])
    
    def ticket_number(x):
        return x.split(" ")[-1]
        
    def ticket_item(x):
        items = x.split(" ")
        if len(items) == 1:
            return "NONE"
        return "_".join(items[0:-1])
    
    def get_title(name):
        title = name.split(",")[1].split(".")[0].strip()
        rare_titles = ["Lady","Countess","Capt","Col","Don","Dr",
                       "Major","Rev","Sir","Jonkheer","Dona"]
        if title in rare_titles:
            return "Rare"
        mapping = {"Mlle":"Miss","Ms":"Miss","Mme":"Mrs"}
        return mapping.get(title, title)
    
    df["Title"] = df["Name"].apply(get_title)
    
    # FamilySize: +1でバグ修正済み（本人を含む人数）
    df["FamilySize"] = df["SibSp"] + df["Parch"] + 1
    #df["IsAlone"] = (df["FamilySize"] == 1).astype(int)
    
    df["Sex_Pclass"] = df["Sex"] + "_" + df["Pclass"].astype(str)
    
    #df["Deck"] = df["Cabin"].astype(str).str[0]
    #df["HasCabin"] = df["Cabin"].notna().astype(int)
    
    #df["FarePerPerson"] = df["Fare"] / df["FamilySize"]
    
    #df["IsChild"] = (df["Age"] < 16).astype(int)
    
    #df["NameLength"] = df["Name"].apply(len)
    
    # Age補完は行わない。TF-DFのネイティブ欠損値処理に任せる
    
    df["Name"] = df["Name"].apply(normalize_name)
    df["Ticket_number"] = df["Ticket"].apply(ticket_number)
    #df["Ticket_item"] = df["Ticket"].apply(ticket_item)
    
    # チケットのグループサイズ（trainとtestで別々に計算せず、まとめて計算する場合は
    # preprocess()の外でconcatしてから計算する必要あり。ここでは単体df内で計算。）
    #ticket_counts = df["Ticket"].value_counts()
    #df["TicketGroupSize"] = df["Ticket"].map(ticket_counts)
    
    return df

    
preprocessed_train_df = preprocess(train_df)
preprocessed_serving_df = preprocess(serving_df)

preprocessed_train_df.head(5)

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked,Title,FamilySize,Sex_Pclass,Ticket_number
0,1,0,3,Braund Mr Owen Harris,male,22.0,1,0,A/5 21171,7.2500,NaN,S,Mr,2,male_3,21171
1,2,1,1,Cumings Mrs John Bradley Florence Briggs Thayer,female,38.0,1,0,PC 17599,71.2833,C85,C,Mrs,2,female_1,17599
2,3,1,3,Heikkinen Miss Laina,female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S,Miss,1,female_3,3101282
3,4,1,1,Futrelle Mrs Jacques Heath Lily May Peel,female,35.0,1,0,113803,53.1000,C123,S,Mrs,2,female_1,113803
4,5,0,3,Allen Mr William Henry,male,35.0,0,0,373450,8.0500,NaN,S,Mr,1,male_3,373450


Let's keep the list of the input features of the model. Notably, we don't want to train our model on the "PassengerId" and "Ticket" features.

In [93]:
input_features = list(preprocessed_train_df.columns)
input_features.remove("Ticket")
input_features.remove("PassengerId")
input_features.remove("Survived")
#input_features.remove("Ticket_number")
input_features.remove("Embarked")     
input_features.remove("Cabin")  
input_features.remove("Sex")  
input_features.remove("Pclass")  
input_features.remove("SibSp")  
input_features.remove("Parch")  
input_features.remove("Fare") 

print(f"Input features: {input_features}")

Input features: ['Name', 'Age', 'Title', 'FamilySize', 'Sex_Pclass', 'Ticket_number']


# Convert Pandas dataset to TensorFlow Dataset

In [94]:
def tokenize_names(features, labels=None):
    """Divite the names into tokens. TF-DF can consume text tokens natively."""
    features["Name"] =  tf.strings.split(features["Name"])
    return features, labels

train_ds = tfdf.keras.pd_dataframe_to_tf_dataset(preprocessed_train_df,label="Survived").map(tokenize_names)
serving_ds = tfdf.keras.pd_dataframe_to_tf_dataset(preprocessed_serving_df).map(tokenize_names)

# Train model with default parameters

### Train model

First, we are training a GradientBoostedTreesModel model with the default parameters.

In [86]:
model = tfdf.keras.GradientBoostedTreesModel(
    verbose=0, # Very few logs
    features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
    exclude_non_specified_features=True, # Only use the features in "features"
    random_seed=1234,
)
model.fit(train_ds)

self_evaluation = model.make_inspector().evaluation()
print(f"Accuracy: {self_evaluation.accuracy} Loss:{self_evaluation.loss}")

[INFO 2026-07-05T06:48:17.337164708+00:00 kernel.cc:1214] Loading model from path /tmp/tmpay_3cajy/model/ with prefix ad43b94456974875
[INFO 2026-07-05T06:48:17.344243481+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2026-07-05T06:48:17.344283138+00:00 kernel.cc:1046] Use fast generic engine


Accuracy: 0.8478260636329651 Loss:0.826004683971405


# Train model with improved default parameters

Now you'll use some specific parameters when creating the GBT model

In [87]:
model = tfdf.keras.GradientBoostedTreesModel(
    verbose=0, # Very few logs
    features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
    exclude_non_specified_features=True, # Only use the features in "features"
    
    #num_trees=2000,
    
    # Only for GBT.
    # A bit slower, but great to understand the model.
    # compute_permutation_variable_importance=True,
    
    # Change the default hyper-parameters
    # hyperparameter_template="benchmark_rank1@v1",
    
    #num_trees=1000,
    #tuner=tuner
    
    min_examples=1,
    categorical_algorithm="RANDOM",
    #max_depth=4,
    shrinkage=0.05,
    #num_candidate_attributes_ratio=0.2,
    split_axis="SPARSE_OBLIQUE",
    sparse_oblique_normalization="MIN_MAX",
    sparse_oblique_num_projections_exponent=2.0,
    num_trees=2000,
    #validation_ratio=0.0,
    random_seed=1234,
    
)
model.fit(train_ds)

self_evaluation = model.make_inspector().evaluation()
print(f"Accuracy: {self_evaluation.accuracy} Loss:{self_evaluation.loss}")

[INFO 2026-07-05T06:48:23.999918396+00:00 kernel.cc:1214] Loading model from path /tmp/tmpoh3d250d/model/ with prefix 597a927f016147a6
[INFO 2026-07-05T06:48:24.011844239+00:00 decision_forest.cc:661] Model loaded with 57 root(s), 2689 node(s), and 6 input feature(s).
[INFO 2026-07-05T06:48:24.011892885+00:00 kernel.cc:1046] Use fast generic engine


Accuracy: 0.8369565010070801 Loss:0.8859865069389343


Let's look at the model and you can also notice the information about variable importance that the model figured out

In [88]:
model.summary()

Model: "gradient_boosted_trees_model_138"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
Total params: 1
Trainable params: 0
Non-trainable params: 1
_________________________________________________________________
Type: "GRADIENT_BOOSTED_TREES"
Task: CLASSIFICATION
Label: "__LABEL"

Input Features (6):
	Age
	FamilySize
	Name
	Sex_Pclass
	Ticket_number
	Title

No weights

Variable Importance: INV_MEAN_MIN_DEPTH:
    1.    "Sex_Pclass"  0.699823 ################
    2.         "Title"  0.277071 ##
    3.           "Age"  0.273924 ##
    4.          "Name"  0.252447 #
    5. "Ticket_number"  0.200686 

Variable Importance: NUM_AS_ROOT:
    1. "Sex_Pclass" 45.000000 ################
    2.      "Title"  9.000000 ##
    3.       "Name"  3.000000 

Variable Importance: NUM_NODES:
    1.           "Age" 718.000000 ################
    2.          "Name" 267.000000 ####
    3.    "Sex_Pclass" 151.000000 #
   

In [89]:
model_with_importance = tfdf.keras.GradientBoostedTreesModel(
    verbose=0,
    features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
    exclude_non_specified_features=True,
    compute_permutation_variable_importance=True,
    random_seed=1234,
)
model_with_importance.fit(train_ds)
model_with_importance.make_inspector().variable_importances()

[INFO 2026-07-05T06:48:42.775869531+00:00 kernel.cc:1214] Loading model from path /tmp/tmpxo8onbe2/model/ with prefix 87bde94b76f64e19
[INFO 2026-07-05T06:48:42.78266812+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2026-07-05T06:48:42.782713378+00:00 kernel.cc:1046] Use fast generic engine


{'MEAN_DECREASE_IN_PRAUC_2_VS_OTHERS': [("Sex_Pclass" (4; #3),
   0.24467950235026548),
  ("FamilySize" (1; #1), 0.029470134853345042),
  ("Name" (5; #2), 0.015567992120061103),
  ("Title" (4; #5), 0.012826218588942284),
  ("Age" (1; #0), 0.008675345577893623),
  ("Ticket_number" (4; #4), 0.005742475445284612)],
 'SUM_SCORE': [("Sex_Pclass" (4; #3), 344.10882171344383),
  ("Title" (4; #5), 81.97851053329032),
  ("Name" (5; #2), 80.00543233748495),
  ("Age" (1; #0), 65.52424416274152),
  ("FamilySize" (1; #1), 64.8640637271501),
  ("Ticket_number" (4; #4), 20.853206141968826)],
 'INV_MEAN_MIN_DEPTH': [("Sex_Pclass" (4; #3), 0.4449038161395111),
  ("Name" (5; #2), 0.31708006427826185),
  ("Title" (4; #5), 0.25904490476742253),
  ("Age" (1; #0), 0.25049103795346045),
  ("FamilySize" (1; #1), 0.2080740820122068),
  ("Ticket_number" (4; #4), 0.20742494159075964)],
 'MEAN_DECREASE_IN_AP_2_VS_OTHERS': [("Sex_Pclass" (4; #3),
   0.25032251501670966),
  ("FamilySize" (1; #1), 0.0299137108513860

# Make predictions

In [90]:
def prediction_to_kaggle_format(model, threshold=0.5):
    proba_survive = model.predict(serving_ds, verbose=0)[:,0]
    return pd.DataFrame({
        "PassengerId": serving_df["PassengerId"],
        "Survived": (proba_survive >= threshold).astype(int)
    })

def make_submission(kaggle_predictions):
    path="/kaggle/working/submission.csv"
    kaggle_predictions.to_csv(path, index=False)
    print(f"Submission exported to {path}")
    
kaggle_predictions = prediction_to_kaggle_format(model)
make_submission(kaggle_predictions)
!head /kaggle/working/submission.csv

Submission exported to /kaggle/working/submission.csv
PassengerId,Survived
892,0
893,0
894,0
895,0
896,1
897,0
898,0
899,0
900,1


# Training a model with hyperparameter tunning

Hyper-parameter tuning is enabled by specifying the tuner constructor argument of the model. The tuner object contains all the configuration of the tuner (search space, optimizer, trial and objective).


In [91]:
tuner = tfdf.tuner.RandomSearch(num_trials=1000)
tuner.choice("min_examples", [2, 5, 7, 10])
tuner.choice("categorical_algorithm", ["CART", "RANDOM"])

local_search_space = tuner.choice("growing_strategy", ["LOCAL"])
local_search_space.choice("max_depth", [3, 4, 5, 6, 8])

global_search_space = tuner.choice("growing_strategy", ["BEST_FIRST_GLOBAL"], merge=True)
global_search_space.choice("max_num_nodes", [16, 32, 64, 128, 256])

#tuner.choice("use_hessian_gain", [True, False])
tuner.choice("shrinkage", [0.02, 0.05, 0.10, 0.15])
tuner.choice("num_candidate_attributes_ratio", [0.2, 0.5, 0.9, 1.0])


tuner.choice("split_axis", ["AXIS_ALIGNED"])
oblique_space = tuner.choice("split_axis", ["SPARSE_OBLIQUE"], merge=True)
oblique_space.choice("sparse_oblique_normalization",
                     ["NONE", "STANDARD_DEVIATION", "MIN_MAX"])
oblique_space.choice("sparse_oblique_weights", ["BINARY", "CONTINUOUS"])
oblique_space.choice("sparse_oblique_num_projections_exponent", [1.0, 1.5])

# Tune the model. Notice the `tuner=tuner`.
tuned_model = tfdf.keras.GradientBoostedTreesModel(tuner=tuner)
tuned_model.fit(train_ds, verbose=0)

tuned_self_evaluation = tuned_model.make_inspector().evaluation()
print(f"Accuracy: {tuned_self_evaluation.accuracy} Loss:{tuned_self_evaluation.loss}")

Use /tmp/tmpg4y8mxqc as temporary training directory


[INFO 2026-07-05T06:51:02.803696123+00:00 kernel.cc:1214] Loading model from path /tmp/tmpg4y8mxqc/model/ with prefix ce56afbde4bb4fda
[INFO 2026-07-05T06:51:02.824934619+00:00 decision_forest.cc:661] Model loaded with 32 root(s), 2134 node(s), and 15 input feature(s).
[INFO 2026-07-05T06:51:02.824979662+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 2026-07-05T06:51:02.82500603+00:00 kernel.cc:1046] Use fast generic engine


Accuracy: 0.8767123222351074 Loss:0.6400823593139648


In the last line in the cell above, you can see the accuracy is higher than previously with default parameters and parameters set by hand.

This is the main idea behing hyperparameter tuning.

For more information you can follow this tutorial: [Automated hyper-parameter tuning](https://www.tensorflow.org/decision_forests/tutorials/automatic_tuning_colab)

# Making an ensemble

Here you'll create 100 models with different seeds and combine their results

This approach removes a little bit the random aspects related to creating ML models

In the GBT creation is used the `honest` parameter. It will use different training examples to infer the structure and the leaf values. This regularization technique trades examples for bias estimates.

In [95]:
predictions = None
num_predictions = 0

for i in range(100):
    print(f"i:{i}")
    # Possible models: GradientBoostedTreesModel or RandomForestModel
    model = tfdf.keras.GradientBoostedTreesModel(
        verbose=0, # Very few logs
        features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
        exclude_non_specified_features=True, # Only use the features in "features"

        #min_examples=1,
        #categorical_algorithm="RANDOM",
        ##max_depth=4,
        #shrinkage=0.05,
        ##num_candidate_attributes_ratio=0.2,
        #split_axis="SPARSE_OBLIQUE",
        #sparse_oblique_normalization="MIN_MAX",
        #sparse_oblique_num_projections_exponent=2.0,
        #num_trees=2000,
        ##validation_ratio=0.0,
        random_seed=i,
        honest=True,
    )
    model.fit(train_ds)
    
    sub_predictions = model.predict(serving_ds, verbose=0)[:,0]
    if predictions is None:
        predictions = sub_predictions
    else:
        predictions += sub_predictions
    num_predictions += 1

predictions/=num_predictions

kaggle_predictions = pd.DataFrame({
        "PassengerId": serving_df["PassengerId"],
        "Survived": (predictions >= 0.5).astype(int)
    })

make_submission(kaggle_predictions)

i:0


[INFO 2026-07-05T06:55:52.710841859+00:00 kernel.cc:1214] Loading model from path /tmp/tmph5_4apyk/model/ with prefix b2f2cabaedde4fed
[INFO 2026-07-05T06:55:52.714649124+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2026-07-05T06:55:52.71469022+00:00 kernel.cc:1046] Use fast generic engine


i:1


[INFO 2026-07-05T06:55:53.807432154+00:00 kernel.cc:1214] Loading model from path /tmp/tmp90yy906r/model/ with prefix 92b06f88402642d2
[INFO 2026-07-05T06:55:53.817236294+00:00 kernel.cc:1046] Use fast generic engine


i:2


[INFO 2026-07-05T06:55:54.820562748+00:00 kernel.cc:1214] Loading model from path /tmp/tmpogo7mrph/model/ with prefix 3e3b41398e4a4e04
[INFO 2026-07-05T06:55:54.827074153+00:00 kernel.cc:1046] Use fast generic engine


i:3


[INFO 2026-07-05T06:55:55.855974264+00:00 kernel.cc:1214] Loading model from path /tmp/tmpto_qzxz5/model/ with prefix 5be4811066d04605
[INFO 2026-07-05T06:55:55.864352761+00:00 kernel.cc:1046] Use fast generic engine


i:4


[INFO 2026-07-05T06:55:56.842087109+00:00 kernel.cc:1214] Loading model from path /tmp/tmp07zj_9cn/model/ with prefix 99a6f4baca3849b8
[INFO 2026-07-05T06:55:56.848598411+00:00 kernel.cc:1046] Use fast generic engine


i:5


[INFO 2026-07-05T06:55:57.83485757+00:00 kernel.cc:1214] Loading model from path /tmp/tmpjby0qn8a/model/ with prefix 744bbdb1eafa4a30
[INFO 2026-07-05T06:55:57.841357715+00:00 kernel.cc:1046] Use fast generic engine


i:6


[INFO 2026-07-05T06:55:58.822363062+00:00 kernel.cc:1214] Loading model from path /tmp/tmphf8j67sy/model/ with prefix 9c40cdbcb3fa428b
[INFO 2026-07-05T06:55:58.82878122+00:00 kernel.cc:1046] Use fast generic engine


i:7


[INFO 2026-07-05T06:56:01.634542067+00:00 kernel.cc:1214] Loading model from path /tmp/tmp0725wj1l/model/ with prefix 4b3869e0abd04ea1
[INFO 2026-07-05T06:56:01.651745313+00:00 kernel.cc:1046] Use fast generic engine


i:8


[INFO 2026-07-05T06:56:02.875775627+00:00 kernel.cc:1214] Loading model from path /tmp/tmpc9s56ap9/model/ with prefix 87be808bb8694554
[INFO 2026-07-05T06:56:02.887931869+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2026-07-05T06:56:02.88798981+00:00 kernel.cc:1046] Use fast generic engine


i:9


[INFO 2026-07-05T06:56:04.066333187+00:00 kernel.cc:1214] Loading model from path /tmp/tmpe7z11chi/model/ with prefix 914a7f5fef2a42bf
[INFO 2026-07-05T06:56:04.079370875+00:00 kernel.cc:1046] Use fast generic engine


i:10


[INFO 2026-07-05T06:56:05.11550573+00:00 kernel.cc:1214] Loading model from path /tmp/tmp5mc94f0o/model/ with prefix 9230c75b2abc4a04
[INFO 2026-07-05T06:56:05.120625515+00:00 kernel.cc:1046] Use fast generic engine


i:11


[INFO 2026-07-05T06:56:06.1815878+00:00 kernel.cc:1214] Loading model from path /tmp/tmp_8x14v9c/model/ with prefix 1df97f188a67429a
[INFO 2026-07-05T06:56:06.190214857+00:00 kernel.cc:1046] Use fast generic engine


i:12


[INFO 2026-07-05T06:56:07.324975481+00:00 kernel.cc:1214] Loading model from path /tmp/tmpjbrv9qny/model/ with prefix fc738369de76416a
[INFO 2026-07-05T06:56:07.337948456+00:00 kernel.cc:1046] Use fast generic engine


i:13


[INFO 2026-07-05T06:56:08.474743414+00:00 kernel.cc:1214] Loading model from path /tmp/tmpjmv876hj/model/ with prefix bab3d1aabc7b4c7f
[INFO 2026-07-05T06:56:08.489019806+00:00 kernel.cc:1046] Use fast generic engine


i:14


[INFO 2026-07-05T06:56:09.487898864+00:00 kernel.cc:1214] Loading model from path /tmp/tmpzc5nso50/model/ with prefix e47aac2573034fef
[INFO 2026-07-05T06:56:09.492385963+00:00 kernel.cc:1046] Use fast generic engine


i:15


[INFO 2026-07-05T06:56:10.418712024+00:00 kernel.cc:1214] Loading model from path /tmp/tmpkh9m0max/model/ with prefix 8f8701c4d15f4be4
[INFO 2026-07-05T06:56:10.422478276+00:00 kernel.cc:1046] Use fast generic engine


i:16


[INFO 2026-07-05T06:56:11.443128246+00:00 kernel.cc:1214] Loading model from path /tmp/tmpqgw06son/model/ with prefix 0b0227ece77048a6
[INFO 2026-07-05T06:56:11.45059267+00:00 kernel.cc:1046] Use fast generic engine


i:17


[INFO 2026-07-05T06:56:12.462990789+00:00 kernel.cc:1214] Loading model from path /tmp/tmp60g5w0gv/model/ with prefix 1e077b3e648e4d6e
[INFO 2026-07-05T06:56:12.471343738+00:00 kernel.cc:1046] Use fast generic engine


i:18


[INFO 2026-07-05T06:56:13.415132681+00:00 kernel.cc:1214] Loading model from path /tmp/tmpwl_6q6n2/model/ with prefix a807f2407e1d4f78
[INFO 2026-07-05T06:56:13.42043734+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2026-07-05T06:56:13.420475495+00:00 kernel.cc:1046] Use fast generic engine


i:19


[INFO 2026-07-05T06:56:14.444955192+00:00 kernel.cc:1214] Loading model from path /tmp/tmp3kajzefv/model/ with prefix 794cf69f101741ce
[INFO 2026-07-05T06:56:14.455417259+00:00 kernel.cc:1046] Use fast generic engine


i:20


[INFO 2026-07-05T06:56:15.448293444+00:00 kernel.cc:1214] Loading model from path /tmp/tmp035det80/model/ with prefix a36a448aff124678
[INFO 2026-07-05T06:56:15.45632199+00:00 kernel.cc:1046] Use fast generic engine


i:21


[INFO 2026-07-05T06:56:16.419198099+00:00 kernel.cc:1214] Loading model from path /tmp/tmp6vecrer5/model/ with prefix 53733a602d3a4482
[INFO 2026-07-05T06:56:16.425896785+00:00 kernel.cc:1046] Use fast generic engine


i:22


[INFO 2026-07-05T06:56:17.421455862+00:00 kernel.cc:1214] Loading model from path /tmp/tmpjbl83p1i/model/ with prefix ede87dcc53b8452c
[INFO 2026-07-05T06:56:17.428821104+00:00 kernel.cc:1046] Use fast generic engine


i:23


[INFO 2026-07-05T06:56:18.583452579+00:00 kernel.cc:1214] Loading model from path /tmp/tmpejjbs_a7/model/ with prefix 655327f5fd214f0b
[INFO 2026-07-05T06:56:18.598317621+00:00 kernel.cc:1046] Use fast generic engine


i:24


[INFO 2026-07-05T06:56:19.673540958+00:00 kernel.cc:1214] Loading model from path /tmp/tmpoi1cio_0/model/ with prefix fb1c22c37ac946a2
[INFO 2026-07-05T06:56:19.682670554+00:00 kernel.cc:1046] Use fast generic engine


i:25


[INFO 2026-07-05T06:56:20.741553507+00:00 kernel.cc:1214] Loading model from path /tmp/tmpqgav0ofj/model/ with prefix 2c1cca6defeb491a
[INFO 2026-07-05T06:56:20.752328698+00:00 kernel.cc:1046] Use fast generic engine


i:26


[INFO 2026-07-05T06:56:21.738781931+00:00 kernel.cc:1214] Loading model from path /tmp/tmp2coyub61/model/ with prefix c43cbbfaccad4b26
[INFO 2026-07-05T06:56:21.745149293+00:00 kernel.cc:1046] Use fast generic engine


i:27


[INFO 2026-07-05T06:56:22.775516776+00:00 kernel.cc:1214] Loading model from path /tmp/tmpyhc2yubr/model/ with prefix 1a02c3ee636a49e6
[INFO 2026-07-05T06:56:22.784691589+00:00 kernel.cc:1046] Use fast generic engine


i:28


[INFO 2026-07-05T06:56:23.72952238+00:00 kernel.cc:1214] Loading model from path /tmp/tmplktuibb5/model/ with prefix 0b2a39d99ef847e7
[INFO 2026-07-05T06:56:23.733389647+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2026-07-05T06:56:23.733428966+00:00 kernel.cc:1046] Use fast generic engine


i:29


[INFO 2026-07-05T06:56:24.800971361+00:00 kernel.cc:1214] Loading model from path /tmp/tmp7ronbshm/model/ with prefix e29f2712f4af4f27
[INFO 2026-07-05T06:56:24.8087889+00:00 kernel.cc:1046] Use fast generic engine


i:30


[INFO 2026-07-05T06:56:25.935847843+00:00 kernel.cc:1214] Loading model from path /tmp/tmpu9op9byq/model/ with prefix 63dabb34c3064d94
[INFO 2026-07-05T06:56:25.947838011+00:00 kernel.cc:1046] Use fast generic engine


i:31


[INFO 2026-07-05T06:56:27.121540772+00:00 kernel.cc:1214] Loading model from path /tmp/tmposokpnqv/model/ with prefix e2d2f7e5d3174afc
[INFO 2026-07-05T06:56:27.134850362+00:00 kernel.cc:1046] Use fast generic engine


i:32


[INFO 2026-07-05T06:56:28.114202939+00:00 kernel.cc:1214] Loading model from path /tmp/tmpcz37731y/model/ with prefix 3131aff1125d4f65
[INFO 2026-07-05T06:56:28.119873706+00:00 kernel.cc:1046] Use fast generic engine


i:33


[INFO 2026-07-05T06:56:29.209940034+00:00 kernel.cc:1214] Loading model from path /tmp/tmp5qz90_2h/model/ with prefix 76d04b702d884ba5
[INFO 2026-07-05T06:56:29.225149935+00:00 kernel.cc:1046] Use fast generic engine


i:34


[INFO 2026-07-05T06:56:30.204917167+00:00 kernel.cc:1214] Loading model from path /tmp/tmpj7sh7kyd/model/ with prefix a6df0b1b3d0e40f5
[INFO 2026-07-05T06:56:30.210536671+00:00 kernel.cc:1046] Use fast generic engine


i:35


[INFO 2026-07-05T06:56:31.305645703+00:00 kernel.cc:1214] Loading model from path /tmp/tmp7cvvp692/model/ with prefix 8c342e6a3ea34601
[INFO 2026-07-05T06:56:31.318283744+00:00 kernel.cc:1046] Use fast generic engine


i:36


[INFO 2026-07-05T06:56:32.379012196+00:00 kernel.cc:1214] Loading model from path /tmp/tmpaem4u1cp/model/ with prefix 287a5ef1f53b4f70
[INFO 2026-07-05T06:56:32.388343423+00:00 kernel.cc:1046] Use fast generic engine


i:37


[INFO 2026-07-05T06:56:33.453825289+00:00 kernel.cc:1214] Loading model from path /tmp/tmp_n8qs2dd/model/ with prefix b107acf7ad784f2f
[INFO 2026-07-05T06:56:33.463925893+00:00 kernel.cc:1046] Use fast generic engine


i:38


[INFO 2026-07-05T06:56:34.440619204+00:00 kernel.cc:1214] Loading model from path /tmp/tmphl8dex3_/model/ with prefix 17fcd9f1ff52451e
[INFO 2026-07-05T06:56:34.446644809+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2026-07-05T06:56:34.44670519+00:00 kernel.cc:1046] Use fast generic engine


i:39


[INFO 2026-07-05T06:56:35.496953571+00:00 kernel.cc:1214] Loading model from path /tmp/tmpycdewkga/model/ with prefix 922bcde769a345c9
[INFO 2026-07-05T06:56:35.505157098+00:00 kernel.cc:1046] Use fast generic engine


i:40


[INFO 2026-07-05T06:56:36.459000173+00:00 kernel.cc:1214] Loading model from path /tmp/tmpll7o8g4q/model/ with prefix 80bb4e013538444b
[INFO 2026-07-05T06:56:36.46382432+00:00 kernel.cc:1046] Use fast generic engine


i:41


[INFO 2026-07-05T06:56:37.412168286+00:00 kernel.cc:1214] Loading model from path /tmp/tmp7m8rw7sx/model/ with prefix 917931a2c7b34efa
[INFO 2026-07-05T06:56:37.416872096+00:00 kernel.cc:1046] Use fast generic engine


i:42


[INFO 2026-07-05T06:56:38.418834616+00:00 kernel.cc:1214] Loading model from path /tmp/tmpx70p6qjy/model/ with prefix 6939cfa1ae5c47a0
[INFO 2026-07-05T06:56:38.425143276+00:00 kernel.cc:1046] Use fast generic engine


i:43


[INFO 2026-07-05T06:56:39.517985849+00:00 kernel.cc:1214] Loading model from path /tmp/tmpl9a3irtg/model/ with prefix e551679d5c2c4d7e
[INFO 2026-07-05T06:56:39.526546861+00:00 kernel.cc:1046] Use fast generic engine


i:44


[INFO 2026-07-05T06:56:40.518170749+00:00 kernel.cc:1214] Loading model from path /tmp/tmpx9fsj3__/model/ with prefix b6cb88e9e4e54f73
[INFO 2026-07-05T06:56:40.523111721+00:00 kernel.cc:1046] Use fast generic engine


i:45


[INFO 2026-07-05T06:56:42.604134798+00:00 kernel.cc:1214] Loading model from path /tmp/tmpmwc4qtt5/model/ with prefix 9c78b17cf0d643ec
[INFO 2026-07-05T06:56:42.614854683+00:00 kernel.cc:1046] Use fast generic engine


i:46


[INFO 2026-07-05T06:56:43.874444468+00:00 kernel.cc:1214] Loading model from path /tmp/tmpc6u6l7fg/model/ with prefix 0f2bdb0f25a54539
[INFO 2026-07-05T06:56:43.884377442+00:00 kernel.cc:1046] Use fast generic engine


i:47


[INFO 2026-07-05T06:56:44.996979044+00:00 kernel.cc:1214] Loading model from path /tmp/tmpnu5tcctj/model/ with prefix dc2d77a7030f4a55
[INFO 2026-07-05T06:56:45.005211486+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2026-07-05T06:56:45.005288414+00:00 kernel.cc:1046] Use fast generic engine


i:48


[INFO 2026-07-05T06:56:46.118747322+00:00 kernel.cc:1214] Loading model from path /tmp/tmpk0lophv7/model/ with prefix 28b346b5049e4e0c
[INFO 2026-07-05T06:56:46.127645017+00:00 kernel.cc:1046] Use fast generic engine


i:49


[INFO 2026-07-05T06:56:47.210563599+00:00 kernel.cc:1214] Loading model from path /tmp/tmpvm0snw15/model/ with prefix 3b4d9f7f1ac84068
[INFO 2026-07-05T06:56:47.219464118+00:00 kernel.cc:1046] Use fast generic engine


i:50


[INFO 2026-07-05T06:56:48.402174105+00:00 kernel.cc:1214] Loading model from path /tmp/tmp1y3bs1ma/model/ with prefix f5a4724fb5654443
[INFO 2026-07-05T06:56:48.416763205+00:00 kernel.cc:1046] Use fast generic engine


i:51


[INFO 2026-07-05T06:56:49.550619362+00:00 kernel.cc:1214] Loading model from path /tmp/tmp7ed7j68m/model/ with prefix b8886c1588584a97
[INFO 2026-07-05T06:56:49.562466099+00:00 kernel.cc:1046] Use fast generic engine


i:52


[INFO 2026-07-05T06:56:50.520318072+00:00 kernel.cc:1214] Loading model from path /tmp/tmphw76_sx9/model/ with prefix 5507ff4a5eb24520
[INFO 2026-07-05T06:56:50.526395768+00:00 kernel.cc:1046] Use fast generic engine


i:53


[INFO 2026-07-05T06:56:51.5056014+00:00 kernel.cc:1214] Loading model from path /tmp/tmpyi0z5ol7/model/ with prefix adbea8da8ae74492
[INFO 2026-07-05T06:56:51.512425351+00:00 kernel.cc:1046] Use fast generic engine


i:54


[INFO 2026-07-05T06:56:52.432290924+00:00 kernel.cc:1214] Loading model from path /tmp/tmpcahxlmav/model/ with prefix de519218fd564c84
[INFO 2026-07-05T06:56:52.436935399+00:00 kernel.cc:1046] Use fast generic engine


i:55


[INFO 2026-07-05T06:56:53.418366268+00:00 kernel.cc:1214] Loading model from path /tmp/tmpvu2i5t9a/model/ with prefix 7a44b4e6b6314776
[INFO 2026-07-05T06:56:53.42638119+00:00 kernel.cc:1046] Use fast generic engine


i:56


[INFO 2026-07-05T06:56:54.372104748+00:00 kernel.cc:1214] Loading model from path /tmp/tmpu46cf6m9/model/ with prefix dbbef85c648c46e7
[INFO 2026-07-05T06:56:54.377441855+00:00 kernel.cc:1046] Use fast generic engine


i:57


[INFO 2026-07-05T06:56:55.2427079+00:00 kernel.cc:1214] Loading model from path /tmp/tmpc_eh48w6/model/ with prefix 6a1042ebb9414f98
[INFO 2026-07-05T06:56:55.24684674+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2026-07-05T06:56:55.246881603+00:00 kernel.cc:1046] Use fast generic engine


i:58


[INFO 2026-07-05T06:56:56.100500646+00:00 kernel.cc:1214] Loading model from path /tmp/tmpil8klfqd/model/ with prefix 40e964f925e74c47
[INFO 2026-07-05T06:56:56.104903924+00:00 kernel.cc:1046] Use fast generic engine


i:59


[INFO 2026-07-05T06:56:57.125380185+00:00 kernel.cc:1214] Loading model from path /tmp/tmpalkhha8u/model/ with prefix 227edcd8bc064d45
[INFO 2026-07-05T06:56:57.137403114+00:00 kernel.cc:1046] Use fast generic engine


i:60


[INFO 2026-07-05T06:56:58.232214222+00:00 kernel.cc:1214] Loading model from path /tmp/tmp1hcyxaqk/model/ with prefix 5d630452ee824641
[INFO 2026-07-05T06:56:58.24819378+00:00 kernel.cc:1046] Use fast generic engine


i:61


[INFO 2026-07-05T06:56:59.136669084+00:00 kernel.cc:1214] Loading model from path /tmp/tmpffeiwr1i/model/ with prefix 0048889a103a4c9d
[INFO 2026-07-05T06:56:59.141845075+00:00 kernel.cc:1046] Use fast generic engine


i:62


[INFO 2026-07-05T06:57:00.148022207+00:00 kernel.cc:1214] Loading model from path /tmp/tmpg8sspmk8/model/ with prefix 99147d877c144455
[INFO 2026-07-05T06:57:00.158567235+00:00 kernel.cc:1046] Use fast generic engine


i:63


[INFO 2026-07-05T06:57:01.142438021+00:00 kernel.cc:1214] Loading model from path /tmp/tmpwyf_qi5e/model/ with prefix a581e01feb4f491f
[INFO 2026-07-05T06:57:01.152924275+00:00 kernel.cc:1046] Use fast generic engine


i:64


[INFO 2026-07-05T06:57:02.088256961+00:00 kernel.cc:1214] Loading model from path /tmp/tmpbf9qe993/model/ with prefix eb64273eb3354d47
[INFO 2026-07-05T06:57:02.096007946+00:00 kernel.cc:1046] Use fast generic engine


i:65


[INFO 2026-07-05T06:57:03.017581208+00:00 kernel.cc:1214] Loading model from path /tmp/tmp8e4mtifd/model/ with prefix 5005cc0e3fba48ef
[INFO 2026-07-05T06:57:03.023955471+00:00 kernel.cc:1046] Use fast generic engine


i:66


[INFO 2026-07-05T06:57:03.919325648+00:00 kernel.cc:1214] Loading model from path /tmp/tmp6_ffgyy6/model/ with prefix b31c891b99674415
[INFO 2026-07-05T06:57:03.924822993+00:00 kernel.cc:1046] Use fast generic engine


i:67


[INFO 2026-07-05T06:57:04.980602345+00:00 kernel.cc:1214] Loading model from path /tmp/tmpl5zyrtfa/model/ with prefix c1e1df0bf4174516
[INFO 2026-07-05T06:57:04.993787323+00:00 kernel.cc:1046] Use fast generic engine


i:68


[INFO 2026-07-05T06:57:06.241939582+00:00 kernel.cc:1214] Loading model from path /tmp/tmp_jcxt59u/model/ with prefix 4da832dc634143bd
[INFO 2026-07-05T06:57:06.264333151+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2026-07-05T06:57:06.264379329+00:00 kernel.cc:1046] Use fast generic engine


i:69


[INFO 2026-07-05T06:57:07.173171139+00:00 kernel.cc:1214] Loading model from path /tmp/tmpbdxwchov/model/ with prefix 1a5eabb09130427a
[INFO 2026-07-05T06:57:07.178606916+00:00 kernel.cc:1046] Use fast generic engine


i:70


[INFO 2026-07-05T06:57:08.121424811+00:00 kernel.cc:1214] Loading model from path /tmp/tmpjap0d8sh/model/ with prefix 477f8e90499a4358
[INFO 2026-07-05T06:57:08.12823389+00:00 kernel.cc:1046] Use fast generic engine


i:71


[INFO 2026-07-05T06:57:09.031631699+00:00 kernel.cc:1214] Loading model from path /tmp/tmp46k92c95/model/ with prefix 4347b566b84c4b48
[INFO 2026-07-05T06:57:09.03669295+00:00 kernel.cc:1046] Use fast generic engine


i:72


[INFO 2026-07-05T06:57:09.927352545+00:00 kernel.cc:1214] Loading model from path /tmp/tmps9ulu2os/model/ with prefix b0442f35b0084881
[INFO 2026-07-05T06:57:09.931725233+00:00 kernel.cc:1046] Use fast generic engine


i:73


[INFO 2026-07-05T06:57:10.792997414+00:00 kernel.cc:1214] Loading model from path /tmp/tmpfy47q9bh/model/ with prefix d240861bc8fd4ae5
[INFO 2026-07-05T06:57:10.797083852+00:00 kernel.cc:1046] Use fast generic engine


i:74


[INFO 2026-07-05T06:57:11.857036432+00:00 kernel.cc:1214] Loading model from path /tmp/tmpy54p4iet/model/ with prefix fe2518ad5b1447f5
[INFO 2026-07-05T06:57:11.867934862+00:00 kernel.cc:1046] Use fast generic engine


i:75


[INFO 2026-07-05T06:57:12.827596361+00:00 kernel.cc:1214] Loading model from path /tmp/tmplaedpv03/model/ with prefix 5122bf6432354eb3
[INFO 2026-07-05T06:57:12.835771761+00:00 kernel.cc:1046] Use fast generic engine


i:76


[INFO 2026-07-05T06:57:13.709417887+00:00 kernel.cc:1214] Loading model from path /tmp/tmpdtyfszzs/model/ with prefix eb3bc70a831a481e
[INFO 2026-07-05T06:57:13.713400569+00:00 kernel.cc:1046] Use fast generic engine


i:77


[INFO 2026-07-05T06:57:14.798967878+00:00 kernel.cc:1214] Loading model from path /tmp/tmpe_nhc2in/model/ with prefix 9e7abe805327402d
[INFO 2026-07-05T06:57:14.813250347+00:00 kernel.cc:1046] Use fast generic engine


i:78


[INFO 2026-07-05T06:57:15.749677351+00:00 kernel.cc:1214] Loading model from path /tmp/tmpoyldc3p5/model/ with prefix 71946165d3294e8e
[INFO 2026-07-05T06:57:15.760678377+00:00 kernel.cc:1046] Use fast generic engine


i:79


[INFO 2026-07-05T06:57:16.676334501+00:00 kernel.cc:1214] Loading model from path /tmp/tmp_k93c743/model/ with prefix 60655e98291d499c
[INFO 2026-07-05T06:57:16.683643482+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2026-07-05T06:57:16.68368512+00:00 kernel.cc:1046] Use fast generic engine


i:80


[INFO 2026-07-05T06:57:17.811946809+00:00 kernel.cc:1214] Loading model from path /tmp/tmpil9eyy3t/model/ with prefix 93ec1a3fac8f4a0a
[INFO 2026-07-05T06:57:17.828350034+00:00 kernel.cc:1046] Use fast generic engine


i:81


[INFO 2026-07-05T06:57:18.779523574+00:00 kernel.cc:1214] Loading model from path /tmp/tmp9xvj5t9y/model/ with prefix c3db2028976446c6
[INFO 2026-07-05T06:57:18.785747411+00:00 kernel.cc:1046] Use fast generic engine


i:82


[INFO 2026-07-05T06:57:19.701609145+00:00 kernel.cc:1214] Loading model from path /tmp/tmp9714w_nr/model/ with prefix b15608675ae24595
[INFO 2026-07-05T06:57:19.706269908+00:00 kernel.cc:1046] Use fast generic engine


i:83


[INFO 2026-07-05T06:57:20.59575196+00:00 kernel.cc:1214] Loading model from path /tmp/tmpwq_6hq50/model/ with prefix 250006f11f9a40e6
[INFO 2026-07-05T06:57:20.601076456+00:00 kernel.cc:1046] Use fast generic engine


i:84


[INFO 2026-07-05T06:57:21.552111852+00:00 kernel.cc:1214] Loading model from path /tmp/tmpe1y32qe5/model/ with prefix d0dd4d5c20aa4d03
[INFO 2026-07-05T06:57:21.560244299+00:00 kernel.cc:1046] Use fast generic engine


i:85


[INFO 2026-07-05T06:57:22.420669456+00:00 kernel.cc:1214] Loading model from path /tmp/tmpl8b82ewa/model/ with prefix de1fa1d167cc4799
[INFO 2026-07-05T06:57:22.424372681+00:00 kernel.cc:1046] Use fast generic engine


i:86


[INFO 2026-07-05T06:57:23.347444042+00:00 kernel.cc:1214] Loading model from path /tmp/tmpw19qvh8_/model/ with prefix c42e8b0ca76741ff
[INFO 2026-07-05T06:57:23.354168581+00:00 kernel.cc:1046] Use fast generic engine


i:87


[INFO 2026-07-05T06:57:25.308161404+00:00 kernel.cc:1214] Loading model from path /tmp/tmpdnxtpx7w/model/ with prefix 777d3baf4aa94d65
[INFO 2026-07-05T06:57:25.32094437+00:00 kernel.cc:1046] Use fast generic engine


i:88


[INFO 2026-07-05T06:57:26.396207971+00:00 kernel.cc:1214] Loading model from path /tmp/tmponbc_3ha/model/ with prefix b6ad537ded4d4fae
[INFO 2026-07-05T06:57:26.405698095+00:00 kernel.cc:1046] Use fast generic engine


i:89


[INFO 2026-07-05T06:57:27.384285862+00:00 kernel.cc:1214] Loading model from path /tmp/tmp8qs1d0jz/model/ with prefix 8aa7987eeb8940b0
[INFO 2026-07-05T06:57:27.390814869+00:00 abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 2026-07-05T06:57:27.390852044+00:00 kernel.cc:1046] Use fast generic engine


i:90


[INFO 2026-07-05T06:57:28.374596331+00:00 kernel.cc:1214] Loading model from path /tmp/tmpq1sk8om3/model/ with prefix 8fdd280a6f4440fd
[INFO 2026-07-05T06:57:28.381365792+00:00 kernel.cc:1046] Use fast generic engine


i:91


[INFO 2026-07-05T06:57:29.501161882+00:00 kernel.cc:1214] Loading model from path /tmp/tmpnwpfbedw/model/ with prefix eab35b6fd60a4fbd
[INFO 2026-07-05T06:57:29.514507979+00:00 kernel.cc:1046] Use fast generic engine


i:92


[INFO 2026-07-05T06:57:30.50174602+00:00 kernel.cc:1214] Loading model from path /tmp/tmpe5ouwkuu/model/ with prefix 5e38abf15566465c
[INFO 2026-07-05T06:57:30.509663642+00:00 kernel.cc:1046] Use fast generic engine


i:93


[INFO 2026-07-05T06:57:31.51900318+00:00 kernel.cc:1214] Loading model from path /tmp/tmpiru9epd_/model/ with prefix 97bb22f4547d441b
[INFO 2026-07-05T06:57:31.528651349+00:00 kernel.cc:1046] Use fast generic engine


i:94


[INFO 2026-07-05T06:57:32.449523495+00:00 kernel.cc:1214] Loading model from path /tmp/tmp966dpior/model/ with prefix edaeeef59d8c449a
[INFO 2026-07-05T06:57:32.455416602+00:00 kernel.cc:1046] Use fast generic engine


i:95


[INFO 2026-07-05T06:57:33.341305542+00:00 kernel.cc:1214] Loading model from path /tmp/tmp5t6nnya8/model/ with prefix c157c26007614717
[INFO 2026-07-05T06:57:33.345272903+00:00 kernel.cc:1046] Use fast generic engine


i:96


[INFO 2026-07-05T06:57:34.416527827+00:00 kernel.cc:1214] Loading model from path /tmp/tmp0b5r7o6x/model/ with prefix 4a6b1eeb534d473e
[INFO 2026-07-05T06:57:34.429377204+00:00 kernel.cc:1046] Use fast generic engine


i:97


[INFO 2026-07-05T06:57:35.354761553+00:00 kernel.cc:1214] Loading model from path /tmp/tmp07p6sc7x/model/ with prefix 8c00bcd0620b4040
[INFO 2026-07-05T06:57:35.360989031+00:00 kernel.cc:1046] Use fast generic engine


i:98


[INFO 2026-07-05T06:57:36.252435913+00:00 kernel.cc:1214] Loading model from path /tmp/tmpi1de4fbw/model/ with prefix d015738100084a1d
[INFO 2026-07-05T06:57:36.256864855+00:00 kernel.cc:1046] Use fast generic engine


i:99


[INFO 2026-07-05T06:57:37.207489632+00:00 kernel.cc:1214] Loading model from path /tmp/tmpspjab87i/model/ with prefix 4366637791314d82
[INFO 2026-07-05T06:57:37.214352114+00:00 kernel.cc:1046] Use fast generic engine


Submission exported to /kaggle/working/submission.csv
